File: feature/search/src/main/java/net/thunderbird/feature/search/api/SearchSpecification.java
package net.thunderbird.feature.search.api;

import android.os.Parcelable;

import net.thunderbird.feature.search.ConditionsTreeNode;


public interface SearchSpecification extends Parcelable {

    /**
     * Get all the uuids of accounts this search acts on.
     * @return Array of uuids.
     */
    String[] getAccountUuids();

    /**
     * Returns the root node of the condition tree accompanying
     * the search.
     *
     * @return Root node of conditions tree.
     */
    ConditionsTreeNode getConditions();
}

File: feature/search/src/main/java/net/thunderbird/feature/search/api/SearchSpecification.kt
package net.thunderbird.feature.search.api

import android.os.Parcelable
import net.thunderbird.feature.search.ConditionsTreeNode

interface SearchSpecification : Parcelable {
    /**
     * Get all the uuids of accounts this search acts on.
     * @return Set of uuids.
     */
    val accountUuids: Set<String>

    /**
     * Returns the root node of the condition tree accompanying
     * the search.
     *
     * @return Root node of conditions tree.
     */
    val conditions: ConditionsTreeNode
}
